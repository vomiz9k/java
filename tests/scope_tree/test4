class Main {
    public static void main() {
        System.out.println(new Class().sum(1, 2, 3, 4));
    }
}

class Class {
    public int sum (int a, int b, int c, int d) {
        new Class().a(a, b, c, d);
        new Class().b(a, b, d);
        new Class().c(a, b);
        new Class().d(a);
        new Class().e();
    }
}

class Class2 extends Class {
    int c;
    int[] d;
    public int proizv(int a, int b) {
        return a * this.d[1];
    }
    public int neg(int a) {
        int b;
        return 0-a;
    }
    public int one() {
        return 1;
    }
}
