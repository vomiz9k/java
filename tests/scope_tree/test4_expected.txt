Main Class Class2

	sum 
		a b c d

	c d proizv neg one
		a b
		a 
			b 


